### PR TITLE
[1.1] Add hash_op implementation

### PR DESCRIPTION
**Pull request by velconia**
This op is for NLP model.

### CODE REVIEW

File: python/paddle/fluid/layers/nn.py [hunk 1]
```diff
@@ -7494,3 +7495,30 @@ def affine_channel(x, scale=None, bias=None, data_layout='NCHW', name=None):
...
     return out
+
+
+def hash(input, hash_size, num_hash=1, name=None):
+    """
+    hash the input
+     Args:
+        input (Variable): The input variable which is a one-hot word.
+        hash_size (int): The space size for hash algorithm.
+        num_hash (int): The times of hash, default 1.
```
Comment (qingqing01) [inline python/paddle/fluid/layers/nn.py:7506]: add name comments in Args

**Review by typhoonzero — DISMISSED** [review]
LG API

**Review by kuke — DISMISSED** [review]
LGTM

**Review by qingqing01 — COMMENTED** [review]
LGTM API

**Review by qingqing01 — APPROVED** [review]
Approved for API.

**Review by typhoonzero — APPROVED** [review]
LG API

File: python/paddle/fluid/layers/nn.py [hunk 2]
```diff
@@ -7494,3 +7495,31 @@ def affine_channel(x, scale=None, bias=None, data_layout='NCHW', name=None):
         attrs={"data_layout": data_layout},
         outputs={"Out": out})
     return out
+
+
+def hash(input, hash_size, num_hash=1, name=None):
+    """
+    hash the input
```
Comment (panyx0718) [inline python/paddle/fluid/layers/nn.py:7502]: Can you give more detail...

File: python/paddle/fluid/layers/nn.py [hunk 3]
```diff
@@ -7494,3 +7495,31 @@ def affine_channel(x, scale=None, bias=None, data_layout='NCHW', name=None):
...
+     Args:
+        input (Variable): The input variable which is a one-hot word.
+        hash_size (int): The space size for hash algorithm.
```
Comment (panyx0718) [inline python/paddle/fluid/layers/nn.py:7505]: what is size? is it bytes, buckets, elements?

Comment (velconia) [reply python/paddle/fluid/layers/nn.py:7505]: Thanks for reviewing, I polished the doc in [here](https://github.com/PaddlePaddle/Paddle/pull/14101)

**Pull request merged**